src/passes/OptimizeInstructions.cpp

```diff
@@ -709,6 +709,12 @@ struct OptimizeInstructions
           store->value = unary->value;
         }
       }
+    } else {
+      if (features.hasBulkMemory()) {
+        if (auto* ret = optimizeBulkMemory(curr)) {
```

Comment:
It would be good to do a `dynCast<MemoryCopy>` here for consistency with the other cases. That would simplify the `optimizeBulkMemory` function as well. The feature check can become an assertion in that case because `MemoryCopy` can only possibly be present when bulk memory is allowed.